prow/plugins/config.go

```diff
@@ -1940,11 +1940,19 @@ func (b *BranchCleaner) IsPreservedBranch(org, repo, branch string) bool {
 		if branch == pb {
 			return true
 		}
+
+		if match, _ := regexp.MatchString(pb, branch); match {
```

Comment:
This is technically backwards incompatible since we could now include more branches than intended.

Reply:
You're correct. That will happen only if regex characters like `*` are in the branch name right, but those cases could be rare. And even then, we are protecting and not deleting the branches, for this feature. So I'm thinking if adding an entire new set of keys would be worth the effort.

Reply:
OK, since this seems unlikely to be a problem in practice and the breaking change behavior is non-destructive, I'm OK with trying this.
If we get any reports of a problem we'd need to revert this change, but that seems unlikely.
/approve

Reply:
Understood, thank you!

Hoping to update the docs as well with a follow up [PR](https://github.com/kubernetes/test-infra/pull/32057)